src/app/providers/provider-form/azuredevops/azuredevops.component.ts

```diff
@@ -29,9 +33,15 @@ export class AzureDevOpsComponent {
     };
 
     this.azureService.azuredevopsPost(providerData)
-      .pipe(
-        finalize(() => this.providersService.triggerCloseForm(this.providerForm)),
-      )
-      .subscribe();
+      .pipe(first())
```

Comment:
Can the `azureService.azuredevopsPost` even send more than 1 event through the pipe? i.e., is this `.pipe(first())` really needed?

Reply:
The intent with the `first()` call was to solve the issue with retained open subscriptions and memory leaks. I think `first()` is a relativly good way to deal with that as it provides automatic unsubsribe on recieve.
If we wanted to improve this with something like a custom pipe or http client for automatic retrys then perhaps we would have to rethink it due to the number of potential events. As is though i think its good. Other thoughts?

Reply:
This is over my head. I'll take your word for it, but cannot contribute much in response. I'm resolving this discussion